lib/hooks/algolia.ts

```diff
@@ -0,0 +1,13 @@
+import { queryAlgolia } from "@lib/services/algolia";
+import { SearchResponse } from "@algolia/client-search";
+import { Book, Course } from "@prisma/client";
+import { useQuery, UseQueryResult } from "react-query";
+
+export function useAlgolia(search: string): UseQueryResult<
+  SearchResponse<{
+    type: string;
+    entry: Book | Course;
+  }>
```

Comment:
Consider refactoring so that this can be `SearchResponse<SearchItem>` and `AutocompleteItem` can be `Hit<SearchItem>`, where `SearchItem` is
```typescript
{ type: "book"; entry: Book } | { type: "course"; entry: CourseWithDept }
```